src/components/Contacts.js

```diff
@@ -0,0 +1,161 @@
+  FormContainer,
+  ContactSection,
+  FormFieldContainer,
+} from './ui';
+//  /////////////////////////////////////////////////////////////////////////////////////////////////
+
+const fields = [
+  {
+    type: 'text',
+    name: 'yourName',
+    component: (props) => <FormInput {...props} />,
+  },
+  {
+    type: 'email',
+    name: 'email',
+    component: (props) => <FormInput {...props} />,
+  },
+  {
+    type: 'tel',
+    name: 'phoneNumber',
+    component: (props) => <FormInput {...props} />,
+  },
+  {
+    type: 'text',
+    name: 'details',
+    component: (props) => <FormTextArea {...props} />,
+  },
+];
+
+const initialFieldValues = {
+  yourName: '',
+  email: '',
+  phoneNumber: '',
+  details: '',
+};
+
+const initialErrors = {
+  yourName: false,
+  email: false,
+  phoneNumber: false,
+  details: false,
+};
+
+const getErrors = (values) => ({
+  yourName: !GC.USER_NAME_REGEX.test(values.yourName),
+  email: !GC.EMAIL_REGEX.test(values.email),
+  phoneNumber: !GC.PHONE_NUMBER_REGEX.test(values.phoneNumber),
+  details: false,
+});
+
+const enhance = compose(
+  withState('fields', 'setFields', { ...initialFieldValues }),
+  withState('errors', 'setErrors', { ...initialErrors }),
+  withState('requestPending', 'setRequestPending', false),
+  withHandlers({
+    handleHideError: (props) => (name) => {
+      const newErrors = props.errors;
+      newErrors[name] = false;
+      props.setErrors(newErrors);
+    },
+    handleFieldChange: (props) => (e, name) => {
+      const field = e.currentTarget;
+      const newFields = props.fields;
+      newFields[name] = field.value;
+      props.setFields(newFields);
+    },
+    handleSubmit: (props) => () => {
+      const errors = getErrors(props.fields);
+      const isInvalid = Object.values(errors).some(err => err === true);
+      if (isInvalid) return props.setErrors(errors);
+      props.setRequestPending(true);
+      // TODO: http request here async sendRequest(props.fields)
+      props.setRequestPending(false);
+      return true;
+    },
+  }),
+);
+
+export const FormField = (props) => (
+  <FormFieldContainer
+    text={
+      props.isInvalid
+        ? H.getLocaleItem(['errorMessages', props.field.name], props.locale)
+        : ''
+    }
+  >
+    {
+      props.field.component({
+        value: props.value,
+        type: props.field.type,
+        isInvalid: props.isInvalid,
+        onFocus: () => props.handleHideError(props.field.name),
+        onChange: (e) => props.handleFieldChange(e, props.field.name),
+        placeholder: H.getLocaleItem([props.field.name], props.locale),
+      })
+    }
+  </FormFieldContainer>
+);
+
+export const Contacts = (props) => (
+  <Fragment>
+    <SectionTitle
+      titleText={H.getLocaleItem(['contactUs'], props.locale)}
+    />
+    <FormContainer>
+      {
+        fields.map((field) => (
+          <FormField
+            field={field}
+            locale={props.locale}
+            value={props.fields[field.name]}
```

Comment:
R.prop(field.name, props.fields)

Reply:
As I mentioned above - not use ramda if not necessary. All fields are inside this component and error will mean we did something wrong